qiskit/algorithms/evolvers/variational/var_qte.py

```diff
@@ -0,0 +1,235 @@
+       https://doi.org/10.22331/q-2019-10-07-191
+    Algorithms that use variational variational_principles to compute a time evolution for a given
+    Hermitian operator (Hamiltonian) and a quantum state.
+    """
+
+    def __init__(
+        self,
+        variational_principle: VariationalPrinciple,
+        ode_function_factory: OdeFunctionFactory,
+        ode_solver_callable: OdeSolver = RK45,
+        lse_solver_callable: Callable[[np.ndarray, np.ndarray], np.ndarray] = partial(
+            np.linalg.lstsq, rcond=1e-2
+        ),
+        expectation: Optional[ExpectationBase] = None,
+        imag_part_tol: float = 1e-7,
+        num_instability_tol: float = 1e-7,
+        quantum_instance: Optional[QuantumInstance] = None,
+    ) -> None:
+        r"""
+        Args:
+            variational_principle: Variational Principle to be used.
+            ode_function_factory: Factory for the ODE function.
+            ode_solver_callable: ODE solver callable that follows a SciPy ``OdeSolver`` interface.
+            lse_solver_callable: Linear system of equations solver that follows a NumPy
+                ``np.linalg.lstsq`` interface.
+            expectation: An instance of ``ExpectationBase`` which defines a method for calculating
+                expectation values of ``EvolutionProblem.aux_operators``.
+            imag_part_tol: Allowed value of an imaginary part that can be neglected if no
+                imaginary part is expected.
+            num_instability_tol: The amount of negative value that is allowed to be
+                rounded up to 0 for quantities that are expected to be
+                non-negative.
+            quantum_instance: Backend used to evaluate the quantum circuit outputs.
+        """
+        super().__init__()
+        self.variational_principle = variational_principle
+        self._quantum_instance = None
+        if quantum_instance is not None:
+            self.quantum_instance = quantum_instance
+        self.expectation = expectation
+        self.ode_function_factory = ode_function_factory
+        self.ode_solver_callable = ode_solver_callable
+        self.lse_solver_callable = lse_solver_callable
+        self.imag_part_tol = imag_part_tol
+        self.num_instability_tol = num_instability_tol
+
+    @property
+    def quantum_instance(self) -> Optional[QuantumInstance]:
+        """Returns quantum instance."""
+        return self._quantum_instance
+
+    @quantum_instance.setter
+    def quantum_instance(self, quantum_instance: Union[QuantumInstance, Backend]) -> None:
+        """Sets quantum_instance"""
+        if not isinstance(quantum_instance, QuantumInstance):
+            quantum_instance = QuantumInstance(quantum_instance)
+
+        self._quantum_instance = quantum_instance
+        self._circuit_sampler = CircuitSampler(
+            quantum_instance, param_qobj=is_aer_provider(quantum_instance.backend)
+        )
+
+    def _evolve_helper(
```

Comment:
Minor comment, but maybe just `_evolve` would be a better name since this actually does the full time evolution. "Helper" sounds like it would do only a small part 😄